Added "aggregate" function to get statistics about metrics.

quentinleguennec opened the pull request:
Hi, this pull request adds a new function to get aggregated statistics about metrics (using the android function "aggregate").

I also updated the README to mention:
```
Add the following line to the end of your "android/gradle.properties" file:
android.jetifier.ignorelist = jackson-core, jackson-databind, jackson-datatype-jsr310, fastdoubleparser
```
This was preventing me from compiling when I tested on my app.

quentinleguennec committed:
Added "aggregate" function to get statistics about metrics.

quentinleguennec commented:
@imdzx Should I assign that to someone for review? Let me know if I can help.

AliAhmad44550 commented:
@quentinleguennec can you Please add the functions for getting Health Connect aggregate data into buckets? There are two types of buckets you can use duration (AggregateGroupByDurationRequest) and period(AggregateGroupByPeriodRequest).

quentinleguennec committed:
Fixed crash on Android 14.

quentinleguennec committed:
Upgraded Google Health Connect version.
Added support for the Privacy Policy button redirect in the Google Health permission Activity.

quentinleguennec committed:
Changed max page size to fix a bug.
Updated README.md.
Ran AGP upgrade assistant.

quentinleguennec committed:
Updated README.md

quentinleguennec committed:
Updated example app to detect when the "privacy policy" button is tapped from within the permission activity.
Updated README.md.
Updated example app.

quentinleguennec committed:
Added code analysis.

quentinleguennec committed:
Ran dart fix and dart format.

nikolaydymura committed:
Implemented fetching exercise route information for ExerciseSessionRecord

quentinleguennec committed:
Now handling getting data after the Health Connect app is freshly installed.
Improved errors handling.
Refactored code.

AliakseiT committed:
Fix Zone Offset Parsing

nikolaydymura committed:
Merge remote-tracking branch 'refs/remotes/quentinleguennec_origin/feature_added_aggregate_function' into feature_added_aggregate_function-merge

# Conflicts:
#	android/src/main/kotlin/dev/duynp/flutter_health_connect/FlutterHealthConnectPlugin.kt
#	lib/src/flutter_health_connect_factory.dart

AliakseiT committed:
Allow records that set HR start/end to the same moment

nikolaydymura committed:
Disabled empty-serialization error

nikolaydymura committed:
Renamed `onExerciseRoutePermissionCallback` to `onExerciseRouteRequestCallback`

nikolaydymura committed:
Added version check guard

nikolaydymura committed:
Fixed error exceptions

nikolaydymura committed:
Fixed error throwing

nikolaydymura committed:
Wrapped exception on new function

AliakseiT committed:
Add logging and wrap string parsing in try catch

quentinleguennec committed:
[4.0.0]
BREAKING: `writeData` now returns a list of the uids of the created Records instead of a boolean.
Fixed `deleteRecordsByIds`, which would delete the records but then throw an error.
Fixed `deleteRecordsByTime`, which would delete the records but then throw an error.
Added documentation.
Improved the Example app to add a button for each `deleteRecordsByIds` and `deleteRecordsByTime`.

quentinleguennec committed:
Ran formatting.

quentinleguennec committed:
Merge branch 'feature_added_aggregate_function' into test_pr

quentinleguennec committed:
[4.1.0]
Fixed issue with timestamps: [#1](https://github.com/quentinleguennec/flutter_health_connect/pull/1) by [AliakseiT](https://github.com/AliakseiT) and [nikolaydymura](https://github.com/nikolaydymura)
Added `getRecordById` [#1](https://github.com/quentinleguennec/flutter_health_connect/pull/1) by [AliakseiT](https://github.com/AliakseiT) and [nikolaydymura](https://github.com/nikolaydymura)
Improved the Example app to add a button for `getRecordById`.

quentinleguennec committed:
Merge pull request #1 from nikolaydymura/feature_added_aggregate_function

Added exercise route fetching and fixed time zones parsing
